client/login/magic-login/handle-emailed-link-form.jsx

```diff
@@ -62,9 +64,22 @@ class HandleEmailedLinkForm extends React.Component {
 		showMagicLoginLinkExpiredPage: PropTypes.func.isRequired,
 	};
 
-	state = {
-		hasSubmitted: false,
-	};
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			hasSubmitted: false,
+		};
+
+		if (
+			isEmpty( props.emailAddress ) ||
+			isEmpty( props.token ) ||
+			isEmpty( props.tokenTime )
+		) {
+			// Required props are really required :)
+			this.props.showMagicLoginLinkExpiredPage();
```

Comment:
I was about to ask why we would want to call that function in the constructor rather than in a lifecycle method but looking at [the documentation](https://facebook.github.io/react/docs/react-component.html#componentwillmount) it seems that's actually recommended:

> This is the only lifecycle hook called on server rendering. Generally, we recommend using the constructor() instead.